src/deadline_worker_agent/worker.py

```diff
@@ -246,6 +246,12 @@ def run(self) -> None:
                                 grace_time=worker_shutdown.grace_time,
                                 fail_message=worker_shutdown.fail_message,
                             )
+                        else:
+                            # If we are here, it's because self._stop.set() was set causing the monitor_ec2_shutdown thread to join.
+                            # The scheduler thread has a longer wait, so let's wake it up so it can join as well.
+                            self._scheduler.shutdown(
+                                fail_message="The Worker recieved a shutdown event."
```

Comment:
Two suggestions:

1.  nit: type `recieved` &rarr; `received`
2.  Can I make a small suggestion here to indicate that the shutdown event was received locally on the host? Because the worker can also receive shutdown events from the service API responses.

Putting these together, how about:

```suggestion
                                fail_message="The Worker received a shutdown event locally from the host machine."
```